torch/_inductor/scheduler.py

```diff
@@ -1435,34 +1433,38 @@ def check_all_pairs(nodes):
         return sorted(possible_fusions, key=self.score_fusion_key, reverse=True)
 
     def will_fusion_create_cycle(self, node1, node2):
-        """Finds whether there's a path from src to dst caused indirectly by fusion"""
+        """
+        Finds whether there's a path from node1 to node2 (or vice-versa)
+        caused indirectly by other fusions.
+        """
 
-        def check(node):
+        def found_path(node):
+            # only fused nodes can introduce new ancestors.
             if isinstance(node, FusedSchedulerNode) and node not in visited:
                 visited.add(node)
-                cond0 = bool(combined_names & node.recursive_predecessors)
-
-                if cond0:
-                    return cond0
-
-                names = node.get_names()
-                shortcut = names.issubset(combined_predecessors)
-
-                if shortcut:
-                    return cond0
+                if node.get_names().issubset(combined_ancestors):
```

Comment:
Note to reviewer: this just reorders things.

`bool(combined_names & node.ancestors)` should be similar cost to `node.get_names().issubset(combined_ancestors)`

Reply:
Further afterthought: `combined_ancestors` is also likely to be in cache.